docs/wiki/guides/current/Magnetometer.md

```diff
@@ -288,61 +302,112 @@ For the most accurate calibration results:
+Magnetic interference from nearby motors can affect calibration very badly.
+
+If the motors are turned slowly, and you see twitching of the quad icon on the main Configurator page, significant changes in the Mag heading value, or twitches in the Mag data in the sensors tab, you may have a problem. In flight, these offsets should average out, but during calibration, they will not. This can lead to strangely different cal values on repeated testing, no matter how careful you are. An average of several cal runs may be most accurate. The further away the motors are, and the further above or below the plane of the motors, the less of a problem this is.
 
-:::note
-Calibration itself **DOES NOT** check that the orientation of the sensor is correct!
 :::
 
+Calibration values, once acquired, typically do not need to be changed much within your local area. They may require updating or checking if you travel a long way from home.
+
 ### Calibration Initiation
 
-The quad must be disarmed. There are two ways to initiate the calibration process:
+It's best to record your previous calibration values before re-calibrating, especially if they were OK before. This can be done with a Preset>Save command, or with `get mag_calibration` in CLI. If the quad has never been calibrated, the values will be `0,0,0`.
+
+The quad must be disarmed to calibrate.
 
-- clicking the `Calibrate Magnetometer` button in Configurator, keeping connected with a long USB cable.
-- using stick commands on a mode 2 radio (be absolutely sure that the quad is disarmed!):
+The 15s delay gives you time to use sticks to initiate the cal process, put the radio down, pick the quad up, and then, once you are ready, you give it a shake, and it starts collecting data.
+
+There are two ways to initiate the calibration process:
+
+- clicking the `Calibrate Magnetometer` button in Configurator, keeping connected with a long USB cable; or
+- using _stick commands_ on a mode 2 radio (be absolutely sure that the quad is disarmed!):
   - right stick straight down (pitch low with roll centered)
   - left stick in the top right corner (throttle high and yaw fully right)
-- you then have 15s in which to 'tap the frame' hard, which starts the calibration process itself
 
-Once the calibration process is activated, the LED on the FC stops flashing.
+If you have a beeper, there will be two short beeps confirming that you initiated the calibration process.
 
-### Calibration Technique
+You then have 15s in which to get ready. When ready, shake the frame at a rate exceeding 350 deg/s to start recording data for the calibration process itself.
+
+When the movement threshold is reached, the LED on the FC goes solid, and the beeper plays a fast 7-beep pattern. You then move the quad around in 3D space so that every axis, at some point in the next 30s, points directly into and out of the field.
```

Comment:
IIRC new code does not need alignment of axes with magnetic field. measurement data spread over 'measurement sphere' (and changing quickly) are necessary. Figure eight movement is usually used.